### PR TITLE
DM-40194: Add fallback for gaps in the template PSF in decorrelation.

### PR DESCRIPTION
**Pull request by isullivan**
If the simple PSF image calculation fails during image decorrelation, compute an average PSF across the template. Note that this will only be used if the science image is convolved during A&L image differencing.

### CODE REVIEW

File: python/lsst/ip/diffim/utils.py [hunk 1]
```diff
@@ -1173,6 +1173,70 @@ def evaluateMeanPsfFwhm(exposure: afwImage.Exposure,
     return np.nanmean(width)
 
 
+def computeAveragePsf(exposure: afwImage.Exposure,
+                      psfExposureBuffer: float, psfExposureGrid: int) -> float:
```
Comment (arunkannawadi) [inline python/lsst/ip/diffim/utils.py:1177]: The return type is incorrect.

File: python/lsst/ip/diffim/utils.py [hunk 2]
```diff
@@ -1173,6 +1173,70 @@ def evaluateMeanPsfFwhm(exposure: afwImage.Exposure,
...
+        The average PSF across the exposure.
+
+    Raises
+    ------
+    ValueError
+        Raised if the PSF cannot be computed at any of the grid points.
+
+    See Also
+    --------
+    evaluateMeanPsfFwhm
```
Comment (arunkannawadi) [inline python/lsst/ip/diffim/utils.py:1204]: Could you also backlink this from `evaluateMeanPsfFwhm`? And while we are modifying the docstring of `evaluateMeanPsfFwhm`, could you change the first line there to say mean instead of median?

File: python/lsst/ip/diffim/imageDecorrelation.py [hunk 3]
```diff
@@ -261,7 +274,7 @@ def run(self, scienceExposure, templateExposure, subtractedExposure, psfMatching
             corr = self.computeDiffimCorrection(kArr, varianceMean, targetVarianceMean)
 
         correctedImage = self.computeCorrectedImage(corr.corrft, subtractedExposure.image.array)
-        correctedPsf = self.computeCorrectedDiffimPsf(corr.corrft, psfImg.array)
```
Comment (arunkannawadi) [inline python/lsst/ip/diffim/imageDecorrelation.py:264]: I'd just keep `psfImg` as `afw.Image` object. If you revert this change, then `computeAveragePsf` must return an object of that type.

**Review by arunkannawadi — APPROVED** [review]

File: python/lsst/ip/diffim/imageDecorrelation.py [hunk 4]
```diff
@@ -27,11 +27,13 @@
 import lsst.geom as geom
 import lsst.meas.algorithms as measAlg
 import lsst.pex.config as pexConfig
+from lsst.pex.exceptions import InvalidParameterError
```
Comment (arunkannawadi) [inline python/lsst/ip/diffim/imageDecorrelation.py:30]: I don't think this repo uses `isort`, but the imports appear sorted already and it might be nice to maintain it by moving this down by one line.

Comment (isullivan) [reply python/lsst/ip/diffim/imageDecorrelation.py:30]: I'm confused. The imports look sorted to me with this line.

Comment (timj) [reply python/lsst/ip/diffim/imageDecorrelation.py:30]: Isort sorts imports before from x import
